code/utils/brain_activation.py

```diff
@@ -0,0 +1,50 @@
+def brain_activation(subject, task_run, condition, TR = 2.5):
+	img = nib.load(path_bold(subject, task_run) + 'filtered_func_data_mni.nii.gz')
+	data = img.get_data()
+
+	vol_shape, n_trs = data.shape[:-1], data.shape[-1]
+	col = 0
+	X = np.ones((n_trs, 10))
+	tr_times = np.arange(0, 30, TR)
+	hrf_at_trs = hrf(tr_times)
+	n_to_remove = len(hrf_at_trs) - 1
+
+	for i in list_every_cond(subject, task_run):
+		neural = events2neural(cond_path(subject, task_run, i), TR, n_trs)
+		convolved = np.convolve(neural, hrf_at_trs)
+		convolved = convolved[:-n_to_remove]
+		X[:, col] = convolved
+		col = col + 1
+
+	mean_data = np.mean(data, axis=-1)
+	masked, mask = median_otsu(mean_data, 2, 1)
+	linear_drift = np.linspace(-1, 1, n_trs)
+	X[:, 8] = linear_drift
+
+#Masking and Modeling Noise
+	in_brain_tcs = data[mask, :]
+	Y = in_brain_tcs.T
+
+	betas = npl.pinv(X).dot(Y)
+	beta_vols = np.zeros(vol_shape + (10,))
+	beta_vols[mask] = betas.T
+
+#Showing Parameter Maps with Color
+	smooth_data = gaussian_filter(data, [2, 2, 2, 0])
+	Y = smooth_data[mask].T
+
+	betas = npl.pinv(X).dot(Y)
+	beta_vols = np.zeros(vol_shape + (10,))
+	beta_vols[mask] = betas.T
+
+	mean_data[~mask] = np.nan
+	beta_vols[~mask] = np.nan
+
+	nice_cmap_values = np.loadtxt('actc.txt')
+	nice_cmap = colors.ListedColormap(nice_cmap_values, 'actc')
+
+	for i in range(mean_data.shape[-1]):
```

Comment:
@imranyousuf exports 90 .pngs in pwd. Should be exported in a newly created folder based off condition label